### PR TITLE
chore: rewrite the Zoe introduction

### LINKED ISSUE #310: Rewrite Zoe Introduction
Rewrite https://agoric.com/documentation/getting-started/intro-zoe.html

Ideally, import the code snippets and test them.

### PR DESCRIPTION
**Pull request by katelynsills**
This PR rewrites and updates the Zoe introduction. More product information is provided, and only necessary details are included.

This PR also adds the ability to import snippets of code and test them. 

Closes #310

### CODE REVIEW

File: main/getting-started/intro-zoe.md [hunk 1]
```diff
@@ -1,358 +1,332 @@
...
+
+**Zoe is easier.**  Writing a smart contract has traditionally meant
+learning a completely new, untried language. And let's hope you don't
+make any mistakes - if you do, your users might lose millions.
+
+Zoe smart contracts, on the other hand, are written in a secure subset
+of JavaScript. Moreover, Zoe handles all of the escrowing of digital
+assets provided by your users and the subsequent payout of digital
+assets to your users. **Your contract, even if it's buggy, can't cause
+your users to lose their assets.**
```
Comment (Chris-Hibbert) [inline main/getting-started/intro-zoe.md:39]: Just an idea.
```suggestion
assets to your users. **Your contract, even if it has bugs due to a library you 
relied on, can't cause your users to lose their assets.**
```

File: main/getting-started/intro-zoe.md [hunk 2]
```diff
@@ -1,358 +1,332 @@
...
+```
+
+Proposals must use Keywords, which are keys that are ASCII and
+capitalized. In this case, the keywords to use [were determined by the
+contract
+code](https://github.com/Agoric/agoric-sdk/blob/23c3f9df56940230e21a16b4861f40197192fdea/packages/zoe/src/contracts/atomicSwap.js#L29):
+Asset and Price.
+
+We said that we are giving 7 simoleans, so we must send 7 simoleans in
+the form of a ERTP payment. ([ERTP
```
Comment (Chris-Hibbert) [inline main/getting-started/intro-zoe.md:149]:
```suggestion
We said that we are giving 7 simoleans, so zoe will ensure that we include 7 simoleans in the 
the form of a ERTP payment. ([ERTP
```

**Review by Chris-Hibbert — COMMENTED** [review]
some ideas.

File: main/getting-started/intro-zoe.md (hunk 1 above)
Comment (katelynsills) [reply main/getting-started/intro-zoe.md:39]: I like this concept, but I think it might be too wordy for this particular sentence. Is there a good way to still include?

Comment (Chris-Hibbert) [reply main/getting-started/intro-zoe.md:39]: This is shorter. Does it work here?

"even if you relied on a buggy contract"

File: snippets/contract-format.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
+// WARNING: THE CODE IN THIS FILE WILL BE SHOWN ON THE AGORIC
+// DOCUMENTATION SITE.
+
+// #region contractFormat
+// @ts-check
+// Checks the types as defined in JSDoc comments
+
+// Add imports here
+
+// Optional: you may wish to use the Zoe helpers in /contractSupport
```
Comment (Chris-Hibbert) [inline snippets/contract-format.js:10]: starting a path with slash means the wrong thing. I use triple-dot.
```suggestion
// Optional: you may wish to use the Zoe helpers in .../contractSupport
```

File: snippets/contract-format.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
+// WARNING: THE CODE IN THIS FILE WILL BE SHOWN ON THE AGORIC
+// DOCUMENTATION SITE.
```
Comment (Chris-Hibbert) [inline snippets/contract-format.js:2]: thanks for adding a contract template. 

I don't think the warning comment is needed,  and it does show up in the expanded "details".

File: snippets/test-intro-zoe.js [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+    give: { Asset: moolaKit.amountMath.make(3) }, // asset: 3 moola
+    want: { Price: simoleanKit.amountMath.make(7) }, // price: 7 simoleans
+  });
+  const alicePayments = harden({
+    Asset: moolaKit.mint.mintPayment(moolaKit.amountMath.make(3)),
+  });
+
+  const atomicSwapFile = `${__dirname}/../node_modules/@agoric/zoe/src/contracts/atomicSwap`;
+
+  // #region bundle
```
Comment (Chris-Hibbert) [inline snippets/test-intro-zoe.js:37]: Rather than having a really flat namespace, would it make sense to prefix the label with the documentation section or page that the code is for?

```suggestion
  // #region intro-zoe-bundle
```

File: main/getting-started/intro-zoe.md (hunk 1 above)
Comment (Chris-Hibbert) [reply main/getting-started/intro-zoe.md:39]: "even if you relied on a buggy library"

File: .wordlist.txt [hunk 6]
```diff
@@ -509,6 +521,7 @@ otherAmountMath
 otherIssuer
 otherMint
 otherPayment
+ourProposal
```
Comment (Chris-Hibbert) [inline .wordlist.txt:524]: Oh, that's too bad that the tags have to be spell-checked as well. Is there any way to exclude some contexts from spelling checks? No? I just thought I'd ask.

File: main/getting-started/intro-zoe.md [hunk 7]
```diff
@@ -1,358 +1,286 @@
...
+
+
+
+Let's imagine that Alice fills in this code with the particulars of
+the [AtomicSwap implementation](https://github.com/Agoric/agoric-sdk/blob/master/packages/zoe/src/contracts/atomicSwap.js). 
+
+To install this particular code, Alice must first bundle it off-chain, meaning
+that the code and its imports are flattened together:
+
+<<< @/snippets/test-intro-zoe.js#importBundleSource
```
Comment (Chris-Hibbert) [inline main/getting-started/intro-zoe.md:227]: This is empty. Was something supposed to go here?

File: main/getting-started/intro-zoe.md [hunk 8]
```diff
@@ -1,358 +1,286 @@
...
+58](https://github.com/Agoric/agoric-sdk/blob/23c3f9df56940230e21a16b4861f40197192fdea/packages/zoe/src/contracts/atomicSwap.js#L58))
+returns a `creatorInvitation` as the result of `startInstance`, so that's what Alice can use to make
+her offer.
+
+<<< @/snippets/test-intro-zoe.js#startInstance
+
+As per the [Atomic Swap contract
+code](https://github.com/Agoric/agoric-sdk/blob/23c3f9df56940230e21a16b4861f40197192fdea/packages/zoe/src/contracts/atomicSwap.js#L50),
+Alice gets an invitation as a result of her offer. It was this
+invitation that she sends to the counter-party.
```
Comment (Chris-Hibbert) [inline main/getting-started/intro-zoe.md:256]: inconsistent tense
```suggestion
Alice gets an invitation as a result of her offer. This
is the invitation she sends to the counter-party.
```

File: main/getting-started/intro-zoe.md [hunk 9]
```diff
@@ -1,358 +1,286 @@
...
+<<< @/snippets/test-intro-zoe.js#aliceOffer
+
+## Two sides of Zoe: Zoe Service and ZCF (Zoe Contract Facet)
+
+You may have noticed that the contract code's `start` method had a
+parameter called `zcf`. This is the Zoe Contract Facet. Zoe actually
+has two sides: the Zoe Service, which we've seen the users interact with, and
+the Zoe Contract Facet (ZCF), which is accessible to the contract
+code. Note that users do not have access to ZCF, but contract code
+does have access to the Zoe Service.
```
Comment (Chris-Hibbert) [inline main/getting-started/intro-zoe.md:267]:
```suggestion
code. Note that users do not have access to ZCF, but contract code
can get access to the Zoe Service.
```

**Review by Chris-Hibbert — APPROVED** [review]
looks good. A few cosmetic suggestions.

File: snippets/test-intro-zoe.js (hunk 5 above)
Comment (katelynsills) [reply snippets/test-intro-zoe.js:37]: I think we shouldn't put that information in the region, since the snippet import itself includes the filename. I think each doc should have a different coe file.

File: .wordlist.txt (hunk 6 above)
Comment (katelynsills) [reply .wordlist.txt:524]: I think there might be. There's an `ignores` configuration setting that is set to `code` and `pre` currently. The spellchecker is currently running on the input markdown, but if I change it to run on the output html, it might ignore those sections successfully. I'll make an issue.

Comment (katelynsills) [reply .wordlist.txt:524]: Issue here: https://github.com/Agoric/documentation/issues/321

File: main/getting-started/intro-zoe.md (hunk 7 above)
Comment (katelynsills) [reply main/getting-started/intro-zoe.md:227]: Oh wow, I think `yarn lint-fix` really messed it up. I checked this before linting, but not after. Thanks, I'll fix.

File: snippets/contract-format.js (hunk 4 above)
Comment (katelynsills) [reply snippets/contract-format.js:2]: Ah, that's a mistake. I need to use the region in the snippet import. Fixed. 

I like the warning, but if it might accidentally show up on the Agoric Documentation site, that's almost worse. Hmm. You think I should take it out even with the fix?

File: snippets/contract-format.js (hunk 3 above)
Comment (katelynsills) [reply snippets/contract-format.js:10]: How about the full path? @agoric/zoe/src/contractSupport

File: main/getting-started/intro-zoe.md (hunk 7 above)
Comment (katelynsills) [reply main/getting-started/intro-zoe.md:227]: Fixed

File: main/getting-started/intro-zoe.md (hunk 8 above)
Comment (katelynsills) [reply main/getting-started/intro-zoe.md:256]: Good catch.

File: main/getting-started/intro-zoe.md (hunk 1 above)
Comment (katelynsills) [reply main/getting-started/intro-zoe.md:39]: I think this is changing the meaning significantly. I'm trying to say, 'even if the contract *you wrote* is buggy, your users are fine" I think bringing up other contracts and libraries is a good point, but it doesn't seem to fit well.

File: main/getting-started/intro-zoe.md (hunk 2 above)
Comment (katelynsills) [reply main/getting-started/intro-zoe.md:148]: Hmm, I think this sounds like Zoe is helping us make the payment, rather than something that we must do and which Zoe will enforce. I didn't mean to say anything about Zoe here, but more that we need to make a payment.

File: snippets/contract-format.js (hunk 4 above)
Comment (Chris-Hibbert) [reply snippets/contract-format.js:2]: I don't think it's necessary at all. Why do you find it helpful? 

Since all our code (at least in the public repositories :wink:) is publicly visible, I'm not sure what the extra exposure is that's worth a warning.

File: snippets/test-intro-zoe.js (hunk 5 above)
Comment (Chris-Hibbert) [reply snippets/test-intro-zoe.js:38]: > each doc should have a different code file.

In that case, my suggestion is unnecessary.

File: main/getting-started/intro-zoe.md (hunk 1 above)
Comment (Chris-Hibbert) [reply main/getting-started/intro-zoe.md:39]: My goal was to seem less accusatory. (Some) People don't immediately accept that they need to protect customers from their own mistakes, but I think the argument that code you rely on may have unnoticed bugs or bugs introduced later is becoming more accepted.

**Review by Chris-Hibbert — APPROVED** [review]

File: main/getting-started/intro-zoe.md [hunk 10]
```diff
@@ -1,358 +1,286 @@
...
 
-This is an introduction to Zoe, Agoric's smart contract framework. 
-The [Zoe Guide](/zoe/guide/) has additional information, 
-including concepts not covered here. 
+## What is Zoe?
 
-You can run different types of smart contracts on Zoe. In this document, 
-for the most part we focus on Agoric’s Atomic Swap demo contract, but 
-sometimes refer to how Zoe works with other contract types.
+Zoe is Agoric's smart contract framework. You should use Zoe to:
```
Comment (tyg) [inline main/getting-started/intro-zoe.md:10]:
```suggestion
Zoe is Agoric's smart contract framework. Use Zoe to:
```

File: main/getting-started/intro-zoe.md [hunk 11]
```diff
@@ -1,358 +1,286 @@
...
 
-## What is Zoe?
+* **Run your code on-chain**
+* **Mint new digital assets**
+* **Credibly trade assets**
+
+Zoe relies on [ERTP](./ertp-introduction.md), our token standard for fungible and non-fungible
+tokens.
```
Comment (tyg) [inline main/getting-started/intro-zoe.md:17]:
```suggestion
Zoe relies on Agoric's [*Electronic Rights Transfer Protocol (ERTP)*](./ertp-introduction.md), our token standard.
```

File: main/getting-started/intro-zoe.md [hunk 12]
```diff
@@ -1,358 +1,286 @@
...
 
-## What is Zoe?
+* **Run your code on-chain**
+* **Mint new digital assets**
+* **Credibly trade assets**
+
+Zoe relies on [ERTP](./ertp-introduction.md), our token standard for fungible and non-fungible
+tokens. 
+
+## Why should I use Zoe?
```
Comment (tyg) [inline main/getting-started/intro-zoe.md:19]:
```suggestion
## Why use Zoe?
```

File: main/getting-started/intro-zoe.md [hunk 13]
```diff
@@ -1,358 +1,286 @@
...
+
+### For Users ###
+
+**Zoe is safer.** Putting digital assets in a smart contract has
+traditionally meant you might lose them. But Zoe guarantees that you
+will either get what you wanted or get a full refund. You will never
+leave a contract empty-handed, even if the smart contract is buggy or
+malicious.
```
Comment (tyg) [inline main/getting-started/intro-zoe.md:27]:
```suggestion
**Zoe is safer.** Traditionally, putting digital assets in a smart contract has
risked losing them. But Zoe guarantees you
get either what you wanted or a full refund of the assets you put in. You will never
leave a contract empty-handed, even if the smart contract is buggy or
malicious. 
```

File: main/getting-started/intro-zoe.md [hunk 14]
```diff
@@ -1,358 +1,286 @@
...
+traditionally meant you might lose them. But Zoe guarantees that you
+will either get what you wanted or get a full refund. You will never
+leave a contract empty-handed, even if the smart contract is buggy or
+malicious. 
+
+### For Developers ###
+
+**Zoe is easier.**  Writing a smart contract has traditionally meant
+learning a completely new, untried language. And let's hope you don't
+make any mistakes - if you do, your users might lose millions.
```
Comment (tyg) [inline main/getting-started/intro-zoe.md:33]:
```suggestion
**Zoe is easier.**  Traditionally, writing a smart contract meant
learning a new, untried language. And don't
make any mistakes - if you do, your users might lose millions.
```

File: main/getting-started/intro-zoe.md [hunk 15]
```diff
@@ -1,358 +1,286 @@
...
+
+Zoe smart contracts, on the other hand, are written in a secure subset
+of JavaScript. Moreover, Zoe handles all of the escrowing of digital
+assets provided by your users and the subsequent payout of digital
+assets to your users. **Your contract, even if it's buggy, can't cause
+your users to lose their assets.**
```
Comment (tyg) [inline main/getting-started/intro-zoe.md:39]:
```suggestion
However, you write Zoe contracts in a secure subset
of JavaScript. Moreover, Zoe automatically escrows all user digital
assets and handles their subsequent payout. **Even a buggy contract can't cause
users to lose their assets.**
```

File: main/getting-started/intro-zoe.md [hunk 16]
```diff
@@ -1,358 +1,286 @@
...
+
+## How do I use Zoe?
```
Comment (tyg) [inline main/getting-started/intro-zoe.md:41]:
```suggestion
## How to use Zoe
```

File: main/getting-started/intro-zoe.md (hunk 1 above)
Comment (katelynsills) [reply main/getting-started/intro-zoe.md:39]: Hmm, but I think some people aren't relying on any libraries or other code, so that argument doesn't work for them. But I do think that most responsible smart contract devs would be worried about a Parity-like problem.

File: snippets/contract-format.js (hunk 4 above)
Comment (katelynsills) [reply snippets/contract-format.js:2]: Ok, sure, I'll take it out

File: main/getting-started/intro-zoe.md (hunk 11 above)
Comment (katelynsills) [reply main/getting-started/intro-zoe.md:17]: I think it's not important to specify the full name in the Zoe intro, since it doesn't help explain anything and we also don't use the full name often. ERTP and Electronic Rights Transfer Protocol are not great names, so I'd rather not spend much time on them.

File: main/getting-started/intro-zoe.md [hunk 17]
```diff
@@ -1,358 +1,286 @@
...
+
+At this point, Zoe burns our invitation and confirms its validity. Zoe
+also escrows all of our payments and represents their value in amounts
+as our `current allocation` in the contract.
+
+### Obtaining a UserSeat
+
+Making an offer as a user gives back a `UserSeat`. This seat
+represents our position in the ongoing contract. We can use this seat
+to do a number of things, including:
```
Comment (tyg) [inline main/getting-started/intro-zoe.md:175]:
```suggestion
Making an offer as a user returns a `UserSeat`. This seat
represents your position in the ongoing contract instance (your "seat at the table"). 
You can use this seat to:
```

File: main/getting-started/intro-zoe.md [hunk 18]
```diff
@@ -1,358 +1,286 @@
...
+
+1. exit the contract
+2. get information about our position such as our current allocation
+3. get our payouts from Zoe.
```
Comment (tyg) [inline main/getting-started/intro-zoe.md:179]:
```suggestion
1. Exit the contract.
2. Get information about our position such as our current allocation.
3. Get our payouts from Zoe.
```

File: main/getting-started/intro-zoe.md [hunk 19]
```diff
@@ -1,358 +1,286 @@
...
+
+Let's check that our offer was successful:
```
Comment (tyg) [inline main/getting-started/intro-zoe.md:181]:
```suggestion
Check that your offer was successful:
```

File: main/getting-started/intro-zoe.md [hunk 20]
```diff
@@ -1,358 +1,286 @@
...
+
+<<< @/snippets/test-intro-zoe.js#offerResult
+
+The AtomicSwap contract gives back this particular message for our
+offer: "The offer has been accepted. Once the contract has been
+completed, please check your payout." Other contracts and offers may
+return something different. The `offerResult` is entirely up to the
+contract.
```
Comment (tyg) [inline main/getting-started/intro-zoe.md:189]:
```suggestion
In response to your offer, the AtomicSwap contract returns the message: "The offer has been accepted. 
Once the contract has been completed, please check your payout." Other contracts and offers may
return something different. The offer's result is entirely up to the contract.
```

File: main/getting-started/intro-zoe.md [hunk 21]
```diff
@@ -1,358 +1,286 @@
...
+
+### Getting the payouts
```
Comment (tyg) [inline main/getting-started/intro-zoe.md:191]:
```suggestion
### Getting payouts
```

File: main/getting-started/intro-zoe.md [hunk 22]
```diff
@@ -1,358 +1,286 @@
...
+
+Because this was the atomic swap contract, the contract is over once
+the second party escrows the correct assets. Let's get our payout of
+moola using the Keyword we used ('Asset'):
```
Comment (tyg) [inline main/getting-started/intro-zoe.md:195]:
```suggestion
Because this was an Atomic Swap contract, it is over once
the second party escrows the correct assets to fulfill the offer and do the swap. Get your payout of
moola with the Keyword your used ('Asset'):
```

File: main/getting-started/intro-zoe.md [hunk 23]
```diff
@@ -1,358 +1,286 @@
...
+
+<<< @/snippets/test-intro-zoe.js#getPayout
+
+Alice receives her payouts too:
```
Comment (tyg) [inline main/getting-started/intro-zoe.md:199]:
```suggestion
Alice also receives her payouts:
```

File: main/getting-started/intro-zoe.md [hunk 24]
```diff
@@ -1,358 +1,286 @@
...
+
+<<< @/snippets/test-intro-zoe.js#alicePayout
+
+## Writing and installing a contract
+
+We've gone over the process of inspecting an invitation, making an
+offer, receiving a `UserSeat` and getting a payout, but how did Alice
+create the contract instance in the first place?
```
Comment (tyg) [inline main/getting-started/intro-zoe.md:207]:
```suggestion
Now that you've seen how to participate in a contract instance, let's look at how
you'd create a contract and its instances.
```

File: main/getting-started/intro-zoe.md [hunk 25]
```diff
@@ -1,358 +1,286 @@
...
+
+Let's pretend that Alice wrote the contract from scratch, even
+AtomicSwap is already written as one of our example contracts ([see
+the full AtomicSwap code
+here](https://github.com/Agoric/agoric-sdk/blob/master/packages/zoe/src/contracts/atomicSwap.js)).
+Note: All Zoe contracts must have a certain format, shown below.
```
Comment (tyg) [inline main/getting-started/intro-zoe.md:213]:
```suggestion
Let's pretend Alice wrote that contract from scratch, even though
AtomicSwap is one of Agoric's example contracts ([see
the full AtomicSwap code
here](https://github.com/Agoric/agoric-sdk/blob/master/packages/zoe/src/contracts/atomicSwap.js)).
Note: All Zoe contracts must have this format:
```

File: main/getting-started/intro-zoe.md [hunk 26]
```diff
@@ -1,358 +1,286 @@
...
+<<< @/snippets/contract-format.js#contractFormat
+:::
+
+
+
+Let's imagine that Alice fills in this code with the particulars of
+the [AtomicSwap implementation](https://github.com/Agoric/agoric-sdk/blob/master/packages/zoe/src/contracts/atomicSwap.js). 
+
+To install this particular code, Alice must first bundle it off-chain, meaning
+that the code and its imports are flattened together:
```
Comment (tyg) [inline main/getting-started/intro-zoe.md:225]:
```suggestion
Alice fills in this code template with [AtomicSwap's particulars](https://github.com/Agoric/agoric-sdk/blob/master/packages/zoe/src/contracts/atomicSwap.js). 
To install this particular code, Alice first must bundle it off-chain, meaning the code and its imports are flattened together:
```

File: main/getting-started/intro-zoe.md [hunk 27]
```diff
@@ -1,358 +1,286 @@
...
+<<< @/snippets/test-intro-zoe.js#bundle
+
+Then Alice must install it on Zoe:
+
+<<< @/snippets/test-intro-zoe.js#install
+
+The return value is an `installation`, which we saw earlier. This is
+an object that identifies a particular piece of code installed on Zoe.
+It can be compared to other installations, and we can call
+`E(atomicSwapInstallation).getBundle()` to read the code itself.
```
Comment (tyg) [inline main/getting-started/intro-zoe.md:238]:
```suggestion
The return value is an `installation`, which we saw earlier. It is
an object identifying a particular piece of code installed on Zoe.
It can be compared to other installations, and you can call
`E(atomicSwapInstallation).getBundle()` to see the code itself.
```

File: main/getting-started/intro-zoe.md [hunk 28]
```diff
@@ -1,358 +1,286 @@
...
+### Creating an instance
+
+Now Alice can use the installation to create a new instance. She must
+also tell Zoe about the ERTP issuers that she wants to use, by
+specifying their role with Keywords. Alice was escrowing moola, so she
+will use the keyword `Asset` to label the `moolaIssuer`. She wanted
+simoleans, so she will use the keyword `Price` to label the `simoleanIssuer`. This particular contract ([see permalink to line
+58](https://github.com/Agoric/agoric-sdk/blob/23c3f9df56940230e21a16b4861f40197192fdea/packages/zoe/src/contracts/atomicSwap.js#L58))
+returns a `creatorInvitation` as the result of `startInstance`, so that's what Alice can use to make
+her offer.
```
Comment (tyg) [inline main/getting-started/intro-zoe.md:249]:
```suggestion
Now Alice uses the installation to create a new instance. She must
also tell Zoe about the ERTP issuers she wants to use, by
specifying their role with Keywords. Alice was escrowing moola, so she uses the keyword `Asset` to label the `moolaIssuer`. She wanted
simoleans, so she uses the keyword `Price` to label the `simoleanIssuer`. When you create a new instance of the contract ([see permalink to line
58](https://github.com/Agoric/agoric-sdk/blob/23c3f9df56940230e21a16b4861f40197192fdea/packages/zoe/src/contracts/atomicSwap.js#L58))
by calling `startInstance()`, it returns a `creatorInvitation`. Alice uses this to make her offer; even the instance's creator needs to have an invitation to it to participate in it.
```

File: main/getting-started/intro-zoe.md [hunk 29]
```diff
@@ -1,358 +1,286 @@
...
+
+<<< @/snippets/test-intro-zoe.js#startInstance
+
+As per the [Atomic Swap contract
```
Comment (tyg) [inline main/getting-started/intro-zoe.md:253]: I think this is a combination of redundant/conflicting with the previous paragraph? Rephrase or combine the two?

File: main/getting-started/intro-zoe.md [hunk 30]
```diff
@@ -1,358 +1,286 @@
...
+code](https://github.com/Agoric/agoric-sdk/blob/23c3f9df56940230e21a16b4861f40197192fdea/packages/zoe/src/contracts/atomicSwap.js#L50),
+Alice gets an invitation as a result of her offer. This
+is the invitation she sends to the counter-party.
+
+<<< @/snippets/test-intro-zoe.js#aliceOffer
+
+## Two sides of Zoe: Zoe Service and ZCF (Zoe Contract Facet)
```
Comment (tyg) [inline main/getting-started/intro-zoe.md:260]:
```suggestion
## Zoe's two sides: Zoe Service and Zoe Contract Facet (ZCF)
```

File: main/getting-started/intro-zoe.md [hunk 31]
```diff
@@ -1,358 +1,286 @@
...
+
+You may have noticed that the contract code's `start` method had a
+parameter called `zcf`. This is the Zoe Contract Facet. Zoe actually
+has two sides: the Zoe Service, which we've seen the users interact with, and
+the Zoe Contract Facet (ZCF), which is accessible to the contract
+code. Note that users do not have access to ZCF, but contract code
+can get access to the Zoe Service.
```
Comment (tyg) [inline main/getting-started/intro-zoe.md:267]:
```suggestion
You may have noticed the contract code's `start` method had a `zcf`
parameter . This is the Zoe Contract Facet. Zoe
has two sides: the Zoe Service, which you've seen users interact with, and
the Zoe Contract Facet (ZCF), which is accessible to the contract
code. Note that users have access to the Zoe service, but do not have access to the ZCF.
While contract code has access to the ZCF *and* can get access to the Zoe Service. 

The overall Zoe API also includes some very useful Zoe Helper methods, which both you and 
contract code can use.
```

File: main/getting-started/intro-zoe.md [hunk 32]
```diff
@@ -1,358 +1,286 @@
...
+
+To learn more about the Zoe Service and Zoe Contract Facet APIs, [see
```
Comment (tyg) [inline main/getting-started/intro-zoe.md:269]:
```suggestion
To learn more about the Zoe Service, Zoe Contract Facet, and Zoe Helper APIs, [see
```

File: main/getting-started/intro-zoe.md [hunk 33]
```diff
@@ -1,358 +1,286 @@
...
+our Zoe API documentation](/zoe/api/).
 
 ## Next steps
 
-If you want to dive deeper into how Zoe works and what you can do, go to the [Zoe Guide](https://agoric.com/documentation/zoe/guide/#what-is-zoe). 
+If you want to dive deeper into how Zoe works and what you can do, go
+to the [Zoe Guide](/zoe/guide/README.md). 
+
+To learn more about the AtomicSwap contract in particular, you can
+[read its documentation](/zoe/guide/contracts/atomic-swap.md) and take [a look at the source code](https://github.com/Agoric/agoric-sdk/blob/master/packages/zoe/src/contracts/atomicSwap.js).
```
Comment (tyg) [inline main/getting-started/intro-zoe.md:278]:
```suggestion
To learn more about the AtomicSwap contract, you can
[read its documentation](/zoe/guide/contracts/atomic-swap.md) and [look at its source code](https://github.com/Agoric/agoric-sdk/blob/master/packages/zoe/src/contracts/atomicSwap.js).
There are several other example contracts for different transaction types in 
the [Contracts folder](/zoe/guide/contracts/)
```

File: main/getting-started/intro-zoe.md [hunk 34]
```diff
@@ -1,358 +1,286 @@
...
+
+To start building Zoe contracts and applications (dapps), follow the
+instructions in [Starting a
+Project](/getting-started/start-a-project.md) after installing the
+prerequisites.
```
Comment (tyg) [inline main/getting-started/intro-zoe.md:283]:
```suggestion
Project](/getting-started/start-a-project.md) after [installing the
prerequisites](/getting-started/before-using-agoric.md).

File: main/getting-started/intro-zoe.md [hunk 35]
```diff
@@ -1,358 +1,286 @@
...
+
+To explore the Zoe Service and Zoe Contract Facet APIs, see the [Zoe
+API documentation here](/zoe/api/).
```
Comment (tyg) [inline main/getting-started/intro-zoe.md:286]:
```suggestion
To explore the Zoe Service, Zoe Contract Facet, and Zoe Helper APIs, see the [Zoe
API documentation](/zoe/api/).
```

**Review by tyg — COMMENTED** [review]
Generally tried to tighten it up and suggest changing "we" to "you".

File: main/getting-started/intro-zoe.md (hunk 11 above)
Comment (tyg) [reply main/getting-started/intro-zoe.md:17]: On the one hand, yes. On the other, it's generally considered good writing practice to always expand an acronym on first usage, and there are some people who get irked if a doc doesn't do so (the always exception is if it's either effectively become a word in its own right (ATM for the clear reason, NAACP for less obvious reasons) or if one can reasonably assume everyone knows what it stands for)

File: main/getting-started/intro-zoe.md [hunk 36]
```diff
@@ -1,358 +1,286 @@
...
+in which one party puts up some digital assets and sends an invitation
+to a second party to complete the exchange. In this case, Alice has
+already escrowed her side and is asking us to pay a particular price
+to receive her digital assets.
+
+### Inspecting an invitation
+
+So we have an invitation, but how do we use it? We can use Zoe to
+inspect the invitation.
+
```
[the remark this answers is not in the document]
Comment (katelynsills) [reply main/getting-started/intro-zoe.md:88]: sounds good

File: main/getting-started/intro-zoe.md (hunk 11 above)
Comment (katelynsills) [reply main/getting-started/intro-zoe.md:17]: Yeah, good point. I suppose there may readers who will feel like they are missing something, even though there is nothing to miss.

File: main/getting-started/intro-zoe.md [hunk 37]
```diff
@@ -1,358 +1,286 @@
...
+that you can use](/zoe/guide/contracts/), including:
+* a [Uniswap
+implementation](/zoe/guide/contracts/multipoolAutoswap.md)
+* a [covered
+call option contract](/zoe/guide/contracts/covered-call.md)
+* an [OTC
+Desk market maker
+contract](https://github.com/Agoric/agoric-sdk/blob/master/packages/zoe/src/contracts/otcDesk.js)
+* contracts for [minting fungible](https://github.com/Agoric/agoric-sdk/blob/master/packages/zoe/src/contracts/mintPayments.js) and [non-fungible tokens](https://github.com/Agoric/agoric-sdk/blob/master/packages/zoe/src/contracts/mintAndSellNFT.js)
+
```
[the remark this answers is not in the document]
Comment (katelynsills) [reply main/getting-started/intro-zoe.md:59]: Gotcha. I removed the first bit about the three ways. I think we do want people to know about the kind of contracts that can be built on Zoe, with links, early. The reader likely knows what Uniswap is, so that we have something like that, that they would be able to build themselves, is meaningful. And, the links are social proof that we're not talking abstractly and promising more than we can deliver. This is important to a dev audience who has been burned on other hyped projects.

"Example Agoric-written contracts for you to look at and use as templates" isn't right from a product perspective. We want to emphasize composability. That means you'll actually use these contracts - not that you will copy and paste and make your own. Using the term `template` gives the wrong idea. Of course, if anyone writes something new, they will look at other contracts, but we don't need to undermine our composability argument.

The Zoe links are correct. If you start with a `/` it starts from the root, which is `main`.

File: main/getting-started/intro-zoe.md [hunk 38]
```diff
@@ -1,358 +1,286 @@
...
+Let's imagine that our friend Alice has sent a `Zoe invitation` to
+[our wallet](/wallet-api.md). This invitation will let us join Alice's
+running contract instance in a particular role. Without an invitation,
+we are not able to join.
+
+Compare this to a smart contract on Ethereum. On Ethereum, the smart
+contract developer must guard against malicious calls and store an
+internal access control list to check whether the message sender is
+allowed to send such a message. Zoe, built on Agoric's [object
```
[the remark this answers is not in the document]
Comment (katelynsills) [reply main/getting-started/intro-zoe.md:70]: Hmm, why? `can send` sounds like whether it is mechanically possible, which gives the wrong impression, since it's more like `may send`. I think `allowed` is correct.

File: main/getting-started/intro-zoe.md [hunk 39]
```diff
@@ -1,358 +1,286 @@
...
+inspect the invitation.
+
+<<< @/snippets/test-intro-zoe.js#details
+
+::: warning Note
+
+E() is part of the Agoric platform and is used to [call methods on
+remote
+objects and receive a promise for the result](/distributed-programming.md#communicating-with-remote-objects-using-e). From our perspective, Zoe is a
+remote object, so we must use E().
```
[the remark this answers is not in the document]
Comment (katelynsills) [reply main/getting-started/intro-zoe.md:94]: Yup, good point

File: main/getting-started/intro-zoe.md [hunk 40]
```diff
@@ -1,358 +1,286 @@
...
+:::
+
+The invitation contained information about the `installation`, the
+installed code object that was used to create the instance. We can use
+object identity comparison to quickly check that the installation is
+one that we recognize. There is no need to compare line-by-line to
+ensure the code is the same.
```
[the remark this answers is not in the document]
Comment (katelynsills) [reply main/getting-started/intro-zoe.md:101]: This sounds good, thanks!

File: main/getting-started/intro-zoe.md [hunk 41]
```diff
@@ -1,358 +1,286 @@
...
+
+Contracts can add their own information to the invitation. In this
+case, the Atomic Swap contract adds information about what is being
+traded: the `asset`, the amount which Alice has escrowed, and the
+`price`, what we must pay to get the asset. Let's imagine that `asset`
+was an amount of 3 moola, and `price` was an amount of 7 simoleans.
+(Moola and simoleans are made-up currencies for the purpose of this
+example.) Amounts are descriptions of digital assets, but have no
+value themselves. [For more on amounts, please see the ERTP
+guide.](/ertp/guide/amounts.html#amounts-values-and-brands)
```
[the remark this answers is not in the document]
Comment (katelynsills) [reply main/getting-started/intro-zoe.md]: Contracts - it's the code level

File: main/getting-started/intro-zoe.md (hunk 39 above)
[the remark this answers is not in the document]
Comment (katelynsills) [reply main/getting-started/intro-zoe.md]: I added some context. It's a little funky, but hopefully gets the job done

File: main/getting-started/intro-zoe.md (hunk 29 above)
Comment (katelynsills) [reply main/getting-started/intro-zoe.md:254]: Nope, this is not covered before, and it's not conflicting. This is talking about the reader's invitation, not Alice's invitation.
